src/packet_reader.rs

```diff
@@ -0,0 +1,152 @@
+use crate::{
+    constants::MIDI_PACKET_SIZE,
+    event_packet::{MidiPacketParsingError, UsbMidiEventPacket},
+};
+use core::convert::TryFrom;
+
+/// Wrapper for parsing/iterating `UsbMidiEventPacket` messages from a slice
+pub struct MidiPacketBufferReader<'a> {
+    inner: core::slice::Chunks<'a, u8>,
+}
```

Comment:
Is there a reason for using a normal struct instead of a tuple struct here?

Reply:
nah, I'll make it one